renku/service/views/datasets.py

```diff
@@ -262,3 +269,50 @@ def create_dataset_view(user, cache):
             'result': DatasetCreateResponse().load(ctx, unknown=EXCLUDE)
         })
     )
+
+
+@use_kwargs(DatasetImportRequest)
+@marshal_with(DatasetImportResponseRPC)
+@header_doc('Import a dataset', tags=(DATASET_BLUEPRINT_TAG, ))
+@dataset_blueprint.route(
+    '/datasets.import',
+    methods=['POST'],
+    provide_automatic_options=False,
+)
+@handle_base_except
+@handle_validation_except
+@requires_cache
+@requires_identity
+def import_dataset_view(user, cache):
+    """Import a dataset view."""
+    ctx = DatasetImportRequest().load(request.json)
+    project = cache.get_project(user, ctx['project_id'])
+    project_path = make_project_path(user, project)
+
+    if not project_path:
+        return jsonify(
+            error={
+                'code': INVALID_PARAMS_ERROR_CODE,
+                'message': 'invalid project_id: {0}'.format(ctx['project_id']),
+            }
+        )
+
+    user_job = {
+        'job_id': uuid.uuid4().hex,
+        'state': USER_JOB_STATE_ENQUEUED,
+    }
+    cache.create_job(user, user_job)
+
+    with enqueue_retry(DATASETS_JOB_QUEUE) as queue:
+        queue.enqueue(
+            dataset_import,
+            user,
+            user_job['job_id'],
+            ctx['project_id'],
+            ctx['dataset_uri'],
+            short_name=ctx.get('short_name'),
+            extract=ctx.get('extract', False),
+            timeout=os.getenv('WORKER_DATASETS_JOBS_TIMEOUT', 1800),
```

Comment:
Consider also setting `ttl` which says how long a job can be queued. Just so we don't have some jobs that are queued forever in some scenarios.